include/ast.hpp

```diff
@@ -220,17 +220,33 @@ struct CompoundStmtNode : public StmtNode {
   std::vector<std::unique_ptr<StmtNode>> stmts;
 };
 
+/// @brief An external definition is an external declaration that is also a
+/// definition of a function(other than an inline definition) or an object.
+struct ExternDeclNode : public AstNode {
```

Comment:
This comment refers to the external definition, but the node is named `ExternDeclNode`, which is confusing. I understand this sentence is from the manual, but since we are not handling inline functions, it's fine to simplify the explanation. We know that a declaration can also be a definition.

```suggestion
/// @brief An external declaration can be a definition of a function or an object.
struct ExternDeclNode : public AstNode {
```